flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/strategy/ConsumedPartitionGroup.java

```diff
@@ -130,4 +138,12 @@ public boolean areAllPartitionsFinished() {
     public ResultPartitionType getResultPartitionType() {
         return resultPartitionType;
     }
+
+    public ConsumerVertexGroup getConsumerVertexGroup() {
+        return checkNotNull(consumerVertexGroup, "ConsumerVertexGroup is not properly set.");
+    }
+
+    public void setConsumerVertexGroup(ConsumerVertexGroup consumerVertexGroup) {
```

Comment:
Maybe add check `this.consumerVertexGroup == null || this.consumerVertexGroup == consumerVertexGroup` here

Reply:
`this. consumerVertexGroup == null` is good enough I think. We do not expect this set method of one group to be invoked multiple times.